python/pyspark/sql/functions.py

```diff
@@ -910,8 +941,8 @@ def weekofyear(col):
     """
     Extract the week number of a given date as integer.
 
-    >>> df = spark.createDataFrame([('2015-04-08',)], ['a'])
-    >>> df.select(weekofyear(df.a).alias('week')).collect()
+    >>> df = spark.createDataFrame([('2015-04-08',)], ['time'])
```

Comment:
Let's use `d` for `DateType` or `datetime.date` similarly with other existing names.